drivers/dma/dma_esp32_gdma.c

```diff
@@ -418,16 +473,27 @@ static int dma_esp32_get_status(const struct device *dev, uint32_t channel,
 		return -EINVAL;
 	}
 
+	memset(status, 0, sizeof(struct dma_status));
+
 	if (dma_channel->dir == DMA_RX) {
 		status->busy = !gdma_ll_rx_is_fsm_idle(data->hal.dev, dma_channel->channel_id);
 		status->dir = PERIPHERAL_TO_MEMORY;
-		status->read_position = dma_channel->desc.dw0.length;
+		desc = (dma_descriptor_t *)gdma_ll_rx_get_current_desc_addr(
+			data->hal.dev, dma_channel->channel_id);
+		if (desc) {
+			status->read_position = desc - dma_channel->desc_list;
```

Comment:
Is this subtraction always safe and can never have a negative result?

Reply:
Good point. In theory it shut be safe. 
But I change the if statement to `if (desc > dma_channel->desc_list)`. 
That way we make sure its not Null and we never get negative results.